### PR TITLE
Fix condition in package reference list and update dotnet cli version from 1.0.4 to 2.0.0 for non-Windows system

### PR DESCRIPTION
**Pull request by Ky7m**

### CODE REVIEW

**Comment by AndreyAkinshin** [comment]
@Ky7m, thanks!

File: src/BenchmarkDotNet.Core/BenchmarkDotNet.Core.csproj [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <Reference Include="System" />
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.1' Or '$(TargetFramework)' == 'netcoreapp1.1' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.1' Or '$(TargetFramework)' == 'netcoreapp2.0' ">
```
Comment (adamsitnik) [inline src/BenchmarkDotNet.Core/BenchmarkDotNet.Core.csproj]: good catch! I have no idea how did it work without it ;)

Comment (Ky7m) [reply src/BenchmarkDotNet.Core/BenchmarkDotNet.Core.csproj]: @adamsitnik, does TargetFrameworkIdentifier in condition can simplify clause, example
https://github.com/aspnet/MusicStore/blob/dev/samples/MusicStore/MusicStore.csproj#L20,L24
what do you think?

Comment (AndreyAkinshin) [reply src/BenchmarkDotNet.Core/BenchmarkDotNet.Core.csproj]: It makes much more sense to me. I think it's a good idea to use `TargetFrameworkIdentifier` everywhere (not only for `.NETCoreApp`, but also for `.NETFramework`).

Comment (adamsitnik) [reply src/BenchmarkDotNet.Core/BenchmarkDotNet.Core.csproj]: @Ky7m very good idea!

Comment (Ky7m) [reply src/BenchmarkDotNet.Core/BenchmarkDotNet.Core.csproj]: Great, let me grab this item and I'll send a PR

**Pull request merged**